src/init.cpp

```diff
@@ -800,6 +831,63 @@ void InitLogging()
  */
 bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
 {
+    // ********************************************************* Step 0: download bootstrap
+    std::string sBootstrap = GetArg("-bootstrap","");
+
+    if(sBootstrap != "")
+    {
+        RemoveConfigFile("bootstrap");
+        uiInterface.InitMessage(strprintf("[BOOTSTRAP] Trying to download from %s", sBootstrap.c_str()));
+        CURL *curl;
+        FILE *fp, *a;
+        CURLcode res;
+
+        curl = curl_easy_init();
+        if (curl) {
+            std::string sDownload = (GetDataDir(false).string() + "/bootstrap_temp.tar");
+            fp = fopen(sDownload.c_str(),"wb");
+            curl_easy_setopt(curl, CURLOPT_URL, sBootstrap.c_str());
+            curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0);
+            curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0);
+            curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, write_data);
+            curl_easy_setopt(curl, CURLOPT_WRITEDATA, fp);
+            curl_easy_setopt(curl, CURLOPT_XFERINFOFUNCTION, xferinfo);
+            curl_easy_setopt(curl, CURLOPT_NOPROGRESS, 0L);
+            res = curl_easy_perform(curl);
+            if (res != CURLE_OK) {
+                curl_easy_cleanup(curl);
+                curl_global_cleanup();
+                return InitError("[BOOTSTRAP] Failed!");
+            } else {
+                curl_easy_cleanup(curl);
+                curl_global_cleanup();
+                int i=fclose(fp);
+                if(i==0) {
+                    uiInterface.InitMessage("[BOOTSTRAP] Downloaded");
+                    a = fopen(sDownload.c_str(), "rb");
+                    if (a == NULL) {
+                        boost::filesystem::remove_all(sDownload);
+                        return InitError("[BOOTSTRAP] Unable to open downloaded file.");
+                    } else {
+                        boost::filesystem::remove_all(GetDataDir(false) / "blocks");
+                        boost::filesystem::remove_all(GetDataDir(false) / "chainstate");
+                        boost::filesystem::remove_all(GetDataDir(false) / "cfund");
+                        bool fOk = untar(a, sDownload.c_str());
+                        fclose(a);
+                        boost::filesystem::remove_all(sDownload);
+                        if(!fOk)
+                            return InitError("[BOOTSTRAP] Could not extract data.");
+                    }
+                } else {
+                    return InitError("[BOOTSTRAP] Failed!");
+                }
+            }
+        } else {
+            return InitError("[BOOTSTRAP] Failed!");
```

Comment:
Should this be more descriptive, as it's not that the bootstrap failed its the lib doesn't exist? so we exit. Thin it should say, Failed as curl is not found

Reply:
if the lib does not exist, the code won't compile.
curl_easy_init( ) is NULL when "something went wrong" as per Curl official documentation (https://curl.haxx.se/libcurl/c/curl_easy_init.html) - hence the lack of definition in the warning text